### DISCUSSION TITLE
Is there a stable C API ?

### DISCUSSION THREAD

**Discussion by ZhangHanDong** [taichi-dev/taichi#4679] [Ideas]
If you have a stable C Api, contributors interested in `taichi` can play with `taichi` in their preferred language, for example implementing the Rust binding library, which is much more interesting to me.

**Comment by k-ye** [comment]
Hi @ZhangHanDong,

Thanks for expressing the interest on the C API & Rust binding! As a start, we'd like to provide C APIs for our AOT feature first: https://github.com/taichi-dev/taichi/issues/3642. This would allow others to use the compiled Taichi kernels outside the Python environment.

However, this won't touch the part around AST/IR building yet, which is what I believe you're looking for?

**Reply by ZhangHanDong** [reply]
Looking forward to C-APIs
